fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSession.java

```diff
@@ -163,8 +166,18 @@ public void persist(final ResourceOperation operation) throws PersistentStorageE
             }
 
             //resolve the mapping between the fedora resource identifier and the associated OCFL object.
-            final FedoraOCFLMapping mapping = fedoraOcflIndex.getMapping(operation.getResourceId());
-
+            final String resourceId = operation.getResourceId();
+            final FedoraOCFLMapping mapping = fedoraOcflIndex.getMapping(resourceId);
+
+            //if no mapping exists, create one
+            if(mapping == null) {
+                if(operation instanceof CreateResourceOperation) {
```

Comment:
Nit picky but the lack of space between `if` and `(` on these two lines makes my brain hurt.